en/docs/learn/cors-migration.md

```diff
@@ -0,0 +1,515 @@
+            SP_APP_ID           INT NOT NULL,
+        
+            PRIMARY KEY (IDN_CORS_ORIGIN_ID, SP_APP_ID)
+        ) ENGINE INNODB;
+        ALTER TABLE IDN_CORS_ASSOCIATION
+            ADD CONSTRAINT ID_IDN_CORS_ORIGIN_ID_FOREIGN_CONSTRAINT
+                FOREIGN KEY (IDN_CORS_ORIGIN_ID)
+                REFERENCES IDN_CORS_ORIGIN (ID)
+                ON DELETE CASCADE
+                ON UPDATE RESTRICT;
+        ALTER TABLE IDN_CORS_ASSOCIATION
+            ADD CONSTRAINT ID_SP_APP_ID_FOREIGN_CONSTRAINT
+                FOREIGN KEY (SP_APP_ID)
+                REFERENCES SP_APP (ID)
+                ON DELETE CASCADE
+                ON UPDATE RESTRICT;
+        
+        CREATE INDEX IDX_IDN_CORS_ASSOCIATION_SP_APP_ID ON IDN_CORS_ASSOCIATION (SP_APP_ID);
+        
+        CREATE INDEX IDX_IDN_CORS_ASSOCIATION_IDN_CORS_ORIGIN_ID ON IDN_CORS_ASSOCIATION (IDN_CORS_ORIGIN_ID);
+        ```
+    
+    ??? abstract "MySQL Cluster"
+        ```
+        INSERT INTO IDN_CONFIG_TYPE (ID, NAME, DESCRIPTION) VALUES
+        ('8ec6dbf1-218a-49bf-bc34-0d2db52d151c', 'CORS_CONFIGURATION', 'A resource type to keep the tenant CORS configurations');
+
+        CREATE TABLE IF NOT EXISTS IDN_CORS_ORIGIN (
+            ID                INT           NOT NULL AUTO_INCREMENT,
+            TENANT_ID         INT           NOT NULL,
+            ORIGIN            VARCHAR(2048) NOT NULL,
+            UUID              CHAR(36)      NOT NULL,
+        
+            PRIMARY KEY (ID),
+            UNIQUE (TENANT_ID, ORIGIN),
+            UNIQUE (UUID)
+        ) ENGINE NDB;
+        
+        CREATE TABLE IF NOT EXISTS IDN_CORS_ASSOCIATION (
+            IDN_CORS_ORIGIN_ID  INT NOT NULL,
+            SP_APP_ID           INT NOT NULL,
+        
+            PRIMARY KEY (IDN_CORS_ORIGIN_ID, SP_APP_ID)
+        ) ENGINE NDB;
+        ALTER TABLE IDN_CORS_ASSOCIATION
+            ADD CONSTRAINT ID_IDN_CORS_ORIGIN_ID_FOREIGN_CONSTRAINT
+                FOREIGN KEY (IDN_CORS_ORIGIN_ID)
+                REFERENCES IDN_CORS_ORIGIN (ID)
+                ON DELETE CASCADE
+                ON UPDATE RESTRICT;
+        ALTER TABLE IDN_CORS_ASSOCIATION
+            ADD CONSTRAINT ID_SP_APP_ID_FOREIGN_CONSTRAINT
+                FOREIGN KEY (SP_APP_ID)
+                REFERENCES SP_APP (ID)
+                ON DELETE CASCADE
+                ON UPDATE RESTRICT;
+
+        CREATE INDEX IDX_IDN_CORS_ASSOCIATION_SP_APP_ID ON IDN_CORS_ASSOCIATION (SP_APP_ID);
+        
+        CREATE INDEX IDX_IDN_CORS_ASSOCIATION_IDN_CORS_ORIGIN_ID ON IDN_CORS_ASSOCIATION (IDN_CORS_ORIGIN_ID);
+        ```
+    
+    ??? abstract "Oracle"
+        ```
+        INSERT INTO IDN_CONFIG_TYPE (ID, NAME, DESCRIPTION) VALUES
+        ('8ec6dbf1-218a-49bf-bc34-0d2db52d151c', 'CORS_CONFIGURATION', 'A resource type to keep the tenant CORS configurations')
+        /
+        
+        CREATE TABLE IF NOT EXISTS IDN_CORS_ORIGIN (
+            ID                INT           NOT NULL AUTO_INCREMENT,
```

Comment:
Auto Increment cannot be used in oracle Also there is no data type as INT

Reply:
Also IF not EXISTS is not supported

Reply:
CREATE TABLE IDNP_CORS_ORIGIN (		
ID                INTEGER           NOT NULL,		
TENANT_ID         INTEGER           NOT NULL,		
ORIGIN            VARCHAR(2048) NOT NULL,		
UUID              CHAR(36)      NOT NULL,			
PRIMARY KEY (ID),		
UNIQUE (TENANT_ID, ORIGIN),		
UNIQUE (UUID)		
);

CREATE SEQUENCE IDNP_CORS_ORIGIN_SEQ START WITH 1 INCREMENT BY 1 NOCACHE;

CREATE OR REPLACE TRIGGER IDNP_CORS_ORIGIN_TRIG
    BEFORE INSERT
        ON IDNP_CORS_ORIGIN
           REFERENCING NEW AS NEW
           FOR EACH ROW
BEGIN
    SELECT IDNP_CORS_ORIGIN_SEQ.nextval INTO :NEW.ID FROM dual;
END;

Reply:
Fixed in the current implementation.